internal/component/database_observability/mysql/collector/explain_plan.go

```diff
@@ -0,0 +1,735 @@
+	if err != nil {
+		level.Error(c.logger).Log("msg", "failed to fetch digests for explain plans", "err", err)
+		return err
+	}
+	defer rs.Close()
+
+	// Populate cache
+	for rs.Next() {
+		if err := rs.Err(); err != nil {
+			level.Error(c.logger).Log("msg", "failed to iterate rs digests for explain plans", "err", err)
+			return err
+		}
+
+		var qi queryInfo
+		if err = rs.Scan(&qi.schemaName, &qi.digest, &qi.queryText); err != nil {
+			level.Error(c.logger).Log("msg", "failed to scan digest for explain plans", "err", err)
+			return err
+		}
+		c.queryCache = append(c.queryCache, qi)
+	}
+	// Calculate batch size based on current cache size
+	c.currentBatchSize = int(math.Ceil(float64(len(c.queryCache)) * c.perScrapeRatio))
+	level.Info(c.logger).Log("msg", "fetched digests", "count", len(c.queryCache), "batch_size", c.currentBatchSize)
+	return nil
+}
+
+func (c *ExplainPlan) fetchExplainPlans(ctx context.Context) error {
+	if len(c.queryCache) == 0 {
+		if err := c.populateQueryCache(ctx); err != nil {
+			return err
+		}
+	}
+
+	processedCount := 0
+	for i, qi := range c.queryCache {
+		if processedCount >= c.currentBatchSize {
+			break
+		}
+		logger := log.With(c.logger, "digest", qi.digest)
+
+		defer func(index int) {
+			c.queryCache = slices.Delete(c.queryCache, index, index+1)
+			processedCount++
+		}(i)
+
+		if strings.HasSuffix(qi.queryText, "...") {
+			level.Debug(logger).Log("msg", "skipping truncated query")
+			continue
+		}
+
+		if !strings.HasPrefix(strings.ToLower(qi.queryText), "select") {
+			continue
+		}
+
+		if qi.schemaName != nil {
+			logger = log.With(logger, "schema_name", *qi.schemaName)
+			if _, err := c.dbConnection.ExecContext(ctx, "USE "+*qi.schemaName); err != nil {
+				level.Error(logger).Log("msg", "failed to set schema", "err", err)
+				continue
+			}
+		}
+
+		rsExplain := c.dbConnection.QueryRowContext(ctx, selectExplainPlanPrefix+qi.queryText)
+
+		if err := rsExplain.Err(); err != nil {
+			level.Error(logger).Log("msg", "failed to run explain plan", "err", err)
+
+			continue
+		}
+
+		var byteExplainPlanJSON []byte
+		if err := rsExplain.Scan(&byteExplainPlanJSON); err != nil {
+			level.Error(logger).Log("msg", "failed to scan explain plan json", "err", err)
+			continue
+		}
+
+		if len(byteExplainPlanJSON) == 0 {
+			level.Error(logger).Log("msg", "explain plan json bytes is empty")
+			continue
+		}
+
+		if !utf8.Valid(byteExplainPlanJSON) {
+			level.Error(logger).Log("msg", "explain plan json bytes is not valid UTF-8")
+			continue
+		}
+
+		loggedSchemaName := "<nil>"
+		if qi.schemaName != nil {
+			loggedSchemaName = *qi.schemaName
+		}
+
+		redactedByteExplainPlanJSON, _, err := redactAttachedConditions(byteExplainPlanJSON)
+		if err != nil {
+			level.Error(logger).Log("msg", "failed to redact explain plan json", "err", err)
+			continue
+		}
+
+		level.Debug(logger).Log("msg", "db native explain plan",
+			"op", OP_EXPLAIN_PLAN_OUTPUT,
```

Comment:
I think we should use a separate `op` here, to avoid issues on the frontend when fetching log lines. (I know we talked this is for on-demand debug purpose, but technically it still need to be a different op)

Reply:
Did do.. It's now `OP_NATIVE_EXPLAIN_PLAN`.

Reply:
> This also get's logged as alloy log output

Ah right, sorry I didn't realise. Just to avoid any possible confusion I think is still good to use a different op (as you've done now).